javascript/lang/security/audit/path-traversal/path-join-resolve-traversal.yaml

```diff
@@ -1,53 +1,68 @@
 rules:
-  - id: path-join-resolve-traversal
-    patterns:
+- id: path-join-resolve-traversal
+  mode: taint
+  pattern-sources:
+  - patterns:
+    - pattern: $X
+    - pattern-either:
+      - pattern-inside: |
+          function ... (...,$X,...) {...}
+      - pattern-inside: |
+          function ... (...,{...,$X,...},...) {...}
+      - pattern-inside: |
+          for ($X of $SMTH) {...}
```

Comment:
how are we feeling about the FP rate of this? Curious if this iterator pattern is disproportionately used for path elements when it's not used on an intrafunction array.

Reply:
yeah, its a good call! if some array config file is used it will be highlighted, I will update the rule :+1: